lang/fastrand/fastrand.go

```diff
@@ -125,26 +125,81 @@ func Uint64n(n uint64) uint64 {
 }
 
 // Read generates len(p) random bytes and writes them into p.
-// It always returns len(p) and a nil error. And it is safe
-// for concurrent use.
-func Read(p []byte) (n int, err error) {
+// It always returns len(p) and a nil error.
+// It is safe for concurrent use.
+func Read(p []byte) (int, error) {
 	l := len(p)
 
+	// Used for local XORSHIFT.
+	var tmp [2]uint32
+	tmp[0], tmp[1] = Uint32(), Uint32()
```

Comment:
We can't do that, here is an optimization. 
If we use `tmp[0], tmp[1] = Uint32(), Uint32()`, the `s0` and `s1` will stay at register instead of writing back to the memory in some codes like `s1 ^= s1 << 17`.